src/util-spm-bm.c

```diff
@@ -267,11 +270,11 @@ static void PreBmGsNocase(const uint8_t *x, uint16_t m, uint16_t *bmGs)
         bmGs[i] = m;
     }
     j = 0;
-    for (i = m - 1; i >= 0; --i) {
-        if (i == -1 || suff[i] == i + 1) {
-            for (; j < m - 1 - i; ++j) {
+    for (i = m; i > 0; --i) {
```

Comment:
this looks like the only logic change, and I have a failing test that only happens with this branch + spm-algo=bm, while it works in master or in this branch with spm-algo=hs

Will share the test offline

Reply:
indeed, I see the bug

Reply:
I changed every usage of `i` by 1 except the one `suff[i]`